### PR TITLE
refactor(proxy): move Context to use Signer

### LINKED ISSUE #753: Parameterize Api on Signer
> In the latest changes of librad, the method of signing was
parameterised. For now, we will fill in this parameter using the
SecretKey. In reality, it should be a parameter itself so we can choose
SecretKey for tests or a client for the main application.

@FintanH in https://github.com/radicle-dev/radicle-upstream/pull/752/commits/4cde468a22fef61f8ab7408f008c73b9a2441aba

### PR DESCRIPTION
**Pull request by xla**
Part of #753

### CODE REVIEW

File: proxy/api/src/http/control.rs [hunk 1]
```diff
@@ -40,16 +40,11 @@ mod handler {
 
     use librad::paths;
 
-<<<<<<< HEAD
```
Comment (rudolfs) [inline proxy/api/src/http/control.rs]: 👀

File: proxy/coco/src/announcement.rs [hunk 2]
```diff
@@ -96,15 +96,19 @@ mod test {
...
 
     #[tokio::test]
     async fn announce() -> Result<(), Error> {
         let tmp_dir = tempfile::tempdir().expect("failed to create temdir");
         let key = SecretKey::new();
-        let config = config::default(key.clone(), tmp_dir.path())?;
+        let signer = signer::BoxedSigner::new(signer::SomeSigner {
+            signer: key.clone(),
+        });
+        let config = config::default(key, tmp_dir.path())?;
```
Comment (FintanH) [inline proxy/coco/src/announcement.rs:108]: Shouldn't the `config` functions take the `BoxedSigner` too? :thinking:

File: proxy/coco/src/peer.rs [hunk 3]
```diff
@@ -774,17 +804,20 @@ mod test {
     #[tokio::test]
     async fn can_clone_user() -> Result<(), Error> {
         let alice_key = SecretKey::new();
+        let alice_signer = signer::BoxedSigner::from(signer::SomeSigner {
```
Comment (FintanH) [inline proxy/coco/src/peer.rs:807]: A quality of life change might be to have `impl From<SecretKey> for BoxedSigner`

File: proxy/coco/src/announcement.rs (hunk 2 above)
Comment (FintanH) [reply proxy/coco/src/announcement.rs:108]: Nvm

**Review by FintanH — APPROVED** [review]

File: proxy/coco/src/announcement.rs (hunk 2 above)
Comment (xla) [reply proxy/coco/src/announcement.rs:108]: For others, this is not part of this change as we still need to land some improvements in link. Until then we at least migrate the majority of the codebase to use signer.

File: proxy/coco/src/peer.rs (hunk 3 above)
Comment (xla) [reply proxy/coco/src/peer.rs:807]: Rusts foreign rules would make that impossible to do in upstream. Do you think this should land in link?

Comment (FintanH) [reply proxy/coco/src/peer.rs:807]: Ya, that's what I meant :) If we notice annoyances in proxy then we should make changes in link if they make sense.

**Review by rudolfs — APPROVED** [review]

**Review by FintanH — APPROVED** [review]

**Pull request merged**